rust/src/memcap.rs

```diff
@@ -0,0 +1,257 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+const BASE: isize = 2;
+
+/// enum to store currently allowed memory units in config,
+/// can be easily expanded for more units
+#[derive(Debug, Eq, PartialEq)]
+pub enum MemUnit {
+    B       = BASE.pow(0),
+    KB      = BASE.pow(10),
+    MB      = BASE.pow(20),
+    GB      = BASE.pow(30),
+    Unknown = 0,
+ }
```

Comment:
hmm.. could probably just define consts for kb, mb and gb on the top and skip these altogether.. Wasn't sure if it would be clean since I was targeting a separate module like `util-misc` for this...